src/index.ts

```diff
@@ -145,6 +146,10 @@ class Schema {
             source = { ...source };
           }
           value = source[z] = rule.transform(value);
+          if (!rule.type) {
+            const type = Array.isArray(value) ? 'array' : typeof value;
+            rule.type = type as RuleType;
+          }
```

Comment:
```suggestion
```suggestion
          const type = Array.isArray(value) ? 'array' : typeof value;
          rule.type ??= rule.type;
```
```

Reply:
done

Reply:
如果 value 是 undefined 类型，会失效，还得改改。